packages/cli-lib/lib/handleFieldsJs.js

```diff
@@ -0,0 +1,218 @@
+        }
+
+        let finalPath = path.join(writeDir, '/fields.json');
+        let json = fieldsArrayToJson(fields);
+        fs.outputFileSync(finalPath, json);
+        logger.info(
+          i18n(`${i18nKey}.converted`, {
+            src: dirName + '/fields.js',
+            dest: dirName + '/fields.json',
+          })
+        );
+        // Switch back to the original directory.
+        process.chdir(cwd);
+        return finalPath;
+      });
+    } catch (e) {
+      logFieldsJsErrors(e, filePath);
+      throw e;
+    }
+  }
+
+  /**
+   * If there has been a fields.json written to the output path, then copy it from the output
+   * directory to the project directory, respecting the path within the output directory.
+   * Ex: path/to/tmp/example.module/fields.json => path/to/project/example.module/fields.output.json
+   */
+  saveOutput() {
+    if (!fs.existsSync(this.outputPath)) {
+      logger.error(
+        `There was an error saving the json output of ${this.filePath}`
+      );
+      return;
+    }
+    const relativePath = path.relative(
+      this.rootWriteDir,
+      path.dirname(this.outputPath)
+    );
+    const savePath = path.join(
+      this.projectDir,
+      relativePath,
+      'fields.output.json'
+    );
+    try {
+      fs.copyFileSync(this.outputPath, savePath);
+    } catch (err) {
+      logger.error(`There was an error saving the json output to ${savePath}`);
+      throw err;
+    }
+  }
+
+  /**
+   * Resolves the relative path to the fields.js within the project directory and returns
+   * directory name to write to in rootWriteDir directory.
+   *
+   * Ex: If rootWriteDir = 'path/to/temp', filePath = 'projectRoot/sample.module/fields.js'. Then getWriteDir() => path/to/temp/sample.module
+   */
+  getWriteDir() {
+    const projectDirRegex = new RegExp(`^${escapeRegExp(this.projectDir)}`);
+    const relativePath = this.filePath.replace(projectDirRegex, '');
+    return path.dirname(path.join(this.rootWriteDir, relativePath));
+  }
+
+  /**
+   * @returns {Promise} Promise that resolves to the path of the output fields.json
+   */
+  getOutputPathPromise() {
+    const writeDir = this.getWriteDir();
+    return this.convertFieldsJs(writeDir).then(outputPath => outputPath);
+  }
+}
+
+//Transform fields array to JSON
+function fieldsArrayToJson(fields) {
+  fields = fields.flat(Infinity).map(field => {
+    return typeof field['toJSON'] === 'function' ? field.toJSON() : field;
+  });
+  return JSON.stringify(fields);
+}
+
+/**
+ * Determines if file is a processable fields.js file (i.e., if it is called 'fields.js' and in a root or in a module folder)
+ */
+function isProcessableFieldsJs(rootDir, filePath) {
+  const regex = new RegExp(`^${escapeRegExp(rootDir)}`);
+  const parts = splitLocalPath(filePath);
+  const relativePath = filePath.replace(regex, '');
+  const baseName = path.basename(filePath);
+  const moduleFolder = parts.find(part => part.endsWith('.module'));
```

Comment:
Might be useful to use https://github.com/HubSpot/hubspot-cli/blob/master/packages/cli-lib/modules.js#L42 here